tests/libreplicaset/replicaset.go

```diff
@@ -13,22 +13,17 @@ import (
 	v1 "kubevirt.io/api/core/v1"
 	"kubevirt.io/client-go/kubecli"
 
-	"kubevirt.io/kubevirt/tests"
 	"kubevirt.io/kubevirt/tests/testsuite"
 )
 
 func DoScaleWithScaleSubresource(virtClient kubecli.KubevirtClient, name string, scale int32) {
 	// Status updates can conflict with our desire to change the spec
 	By(fmt.Sprintf("Scaling to %d", scale))
 	var s *autov1.Scale
-	err := tests.RetryIfModified(func() error {
-		s, err := virtClient.ReplicaSet(testsuite.GetTestNamespace(nil)).GetScale(context.Background(), name, v12.GetOptions{})
-		ExpectWithOffset(1, err).ToNot(HaveOccurred())
-		s.Spec.Replicas = scale
-		s, err = virtClient.ReplicaSet(testsuite.GetTestNamespace(nil)).UpdateScale(context.Background(), name, s)
```

Comment:
Right so the scale subresource that is submitted actually has a status field with the number of observed replicas, which I guess is something that changes quite frequently.. https://github.com/kubevirt/kubevirt/blob/7cd9d125aa9ef7893c40cebd5f38bf765089bb99/vendor/k8s.io/api/autoscaling/v1/types.go#L138-L148

I think we have 2 options:
- Wait for ReplicaSet to be ready/available (ensures observed=desired)
- use PATCH

Reply:
> * Use PATCH

Patch should be used unless one *has* to update the entire object in an atomic fashion. This is not the case here. We should use PATCH.

Reply:
Actually in the scale subresource case (special minimized v1.scale) I think it makes sense we keep the subresource call.

Looks like we just have to set the resourceversion to empty, which is a little icky but being used all over k8s
https://github.com/kubernetes/kubernetes/blob/c3caddf3e696cfcd44c07607254ae6928e85d2f6/test/e2e/apps/deployment.go#L1668-L1674

Reply:
ping wdyt @codingben @dankenigsberg

Reply:
> > * Use PATCH
> 
> Patch should be used unless one _has_ to update the entire object in an atomic fashion. This is not the case here. We should use PATCH.

note that Scale is a sub resource that [does not](https://github.com/kubevirt/kubevirt/blob/81e62b66e79650fbbdfa177b570ab8a4f5c5d086/staging/src/kubevirt.io/client-go/generated/kubevirt/clientset/versioned/typed/core/v1/virtualmachineinstancereplicaset_expansion.go#L33) currently have a patch method.

@codingben perhaps in order to go around to the retry-on-conflict routine - we should add the patch option:
```
type VirtualMachineInstanceReplicaSetExpansion interface {
...
	UpdateScale(ctx context.Context, replicaSetName string, scale *autov1.Scale) (*autov1.Scale, error)
}
```

Reply:
Thanks @RamLavi, I'd like to keep this PR purposed only on removing `RetryIfModified` from `utils.go`. We also can move `RetryIfModified` to `replicaset.go`.

@akalenyu I like your idea by setting `ResourceVersion` to empty. I want to try it, in worst case, we can just keep `RetryIfModified` function in `replicaset.go`.